topsf/tasks/inference_v2/workspace.py

```diff
@@ -0,0 +1,57 @@
+# coding: utf-8
+
+import law
+import os
+
+from topsf.tasks.inference_v2.combine_base import CombineBaseTask
+
+
+class CreateWorkspaceV2(
+    CombineBaseTask,
+):
+    @property
+    def workspace_name(self):
+        name = f"workspace__{law.util.create_hash(self.physics_model)}"
+        return name
+
+    def store_parts(self) -> law.util.InsertableDict:
+        parts = super().store_parts()
+        parts.insert_after("fit_v2", "workspace_v2", self.workspace_name)
+        return parts
+
```

Comment:
Since you already inherit from `FitMixin` the `physics_model` should already be part of the output path via the `fit_name`, so it shouldn't be needed to add this again here.
```suggestion
    @property
    def workspace_name(self):
        name = f"workspace__{law.util.create_hash(self.physics_model)}"
        return name

    def store_parts(self) -> law.util.InsertableDict:
        parts = super().store_parts()
        parts.insert_after("fit_v2", "workspace_v2", self.workspace_name)
        return parts

```

Reply:
Renamed the workspace to ```"workspace__initial"```, indicating that it is the workspace created initially by the text2workspace.py script. We need to check if that workspace file is changed when used as input for the fit tasks. If it is, we should use a copy to work in and not change the initial workspace.